### PR TITLE
[VL] Update CI Python to 3.10 for Spark 4.1 and enable ArrowEvalPythonExecSuite tests

### LINKED ISSUE #11400: [VL] Track on Spark-4.1.x failed unit tests
### Description

After xx the general framework of unit tests are merged. We still have failed unit tests left here:

| Description | Owner | Category | Cause | Affected Files |
|-------------|-------|----------|-------|----------------|
| [Improve `extractShuffleIds` to find `AdaptiveSparkPlanExec` anywhere in plan tree](https://github.com/apache/spark/pull/53620) 4.1.1 | ✔️ @baibaichen  | SQL | [#52157](https://github.com/apache/spark/pull/52157) | `gluten-ut/spark41/.../VeloxTestSettings.scala` |
| Fix `SPARK-47939: Explain should work with parameterized queries` | ✔️ @marin-ma  | TEST | https://github.com/apache/incubator-gluten/pull/11252 |  `gluten-ut/spark41/.../VeloxTestSettings.scala` |
| Support Checksum in Column ShuffleWriters | 🚫 | CORE | [#50230](https://github.com/apache/spark/pull/50230) | gluten-ut/spark41/.../velox/VeloxTestSettings.scala<br>gluten-ut/spark41/.../GlutenMapStatusEndToEndSuite.scala<br>**Excluded tests:**<br>- GlutenMapStatusEndToEndSuite (entire suite) |
| Support `spark.sql.unionOutputPartitioning=true`  | @baibaichen | SQL | [#51623](https://github.com/apache/spark/pull/51623) | `.github/workflows/velox_backend_x86.yml`, `gluten-ut/spark41/.../VeloxTestSettings.scala`, `tools/gluten-it/common/.../Suite.scala`<br>**Excluded tests:**<br>- GlutenBroadcastExchangeSuite.SPARK-52962<br>- GlutenDataFrameSetOperationsSuite.SPARK-52921* |
| Fixes a Spark Parquet read bug where missing struct fields caused the entire struct to be read as NULL. |❓️ | PARQUET | [SPARK-53535](https://issues.apache.org/jira/browse/SPARK-53535) | `gluten-ut/spark41/.../VeloxTestSettings.scala`<br>**Excluded tests:**<br>- SPARK-53535*<br>- vectorized reader: missing all struct fields* |
| Infer Variant shredding schema when writing to Parquet | ❓️ | PARQUET | [#52406](https://github.com/apache/spark/pull/52406) | gluten-ut/spark41/.../velox/VeloxTestSettings.scala<br>**Excluded test:**<br>- "infer shredding with mixed scale" in GlutenFileBasedDataSourceSuite |
| NullType/VOID/UNKNOWN Type Support in Parquet |❓️ | PARQUET | [SPARK-54220](https://issues.apache.org/jira/browse/SPARK-54220) | `gluten-ut/spark41/.../VeloxTestSettings.scala`<br>**Excluded tests:**<br>- SPARK-54220* |
| Update CI python to 3.10 | ❓️ | PYTHON | [#51259](https://github.com/apache/spark/pull/51259) | backends-velox/.../python/ArrowEvalPythonExecSuite.scala |
| Align with Spark `split` | ❓️ | SQL | [#48470](https://github.com/apache/spark/pull/48470) | `gluten-ut/spark41/.../VeloxTestSettings.scala`, `backends-velox/.../VeloxStringFunctionsSuite.scala`<br>**Excluded tests:**<br>- GlutenRegexpExpressionsSuite.SPLIT<br>- VeloxStringFunctionsSuite: split test |
| Fix additional Spark 4.1 `KeyGroupedPartitioningSuite tests` | ❓️ | SQL | [#53132](https://github.com/apache/spark/pull/53132), [#53142](https://github.com/apache/spark/pull/53142) | `gluten-ut/spark41/.../VeloxTestSettings.scala`<br>**Excluded tests:**<br>- SPARK-53322*<br>- SPARK-54439* |
| Fix failing SQL tests on Spark 4.1. | ❓️ | SQL | N/A | gluten-ut/spark41/.../velox/VeloxSQLQueryTestSettings.scala<br>**Excluded tests:**<br>- cast.sql<br>- describe.sql<br>- nonansi/cast.sql<br>- nonansi/st-functions.sql<br>- scripting/randomly_generated_scripts.sql<br>- st-functions.sql<br>- type-coercion-edge-cases.sql<br>- variant-field-extractions.sql |
| Support memory based thresholds for shuffle spill | ❌️ | SQL | [#47856](https://github.com/apache/spark/pull/47856) | `gluten-ut/spark41/.../VeloxTestSettings.scala`<br>**Excluded tests:**<br>- SPARK-49386: Window spill with more than the inMemoryThreshold and spillSizeThreshold<br>- SPARK-49386: test SortMergeJoin (with spill by size threshold) |
| Fix additional Spark 4.1 STRUCTURED STREAMING tests. | ❌️ | SS | [#52645](https://github.com/apache/spark/pull/52645) | `gluten-ut/spark41/.../VeloxTestSettings.scala`<br>**Excluded tests:**<br>- SPARK-53942: changing the number of stateless shuffle partitions via config<br>- SPARK-53942: stateful shuffle partitions are retained from old checkpoint |
| Fix additional Spark 4.1 STRUCTURED STREAMING tests. | ❌️ | SS | [#52473](https://github.com/apache/spark/pull/52473)<br>[#52870](https://github.com/apache/spark/pull/52870)<br>[#52891](https://github.com/apache/spark/pull/52891) | gluten-ut/spark41/.../velox/VeloxTestSettings.scala<br>**Excluded tests:**<br>- GlutenStreamRealTimeModeAllowlistSuite: "rtm operator allowlist", "repartition not allowed", "stateful queries not allowed"<br>- GlutenStreamRealTimeModeE2ESuite: "foreach", "to_json and from_json round-trip", "generateExec passthrough"<br>- GlutenStreamRealTimeModeSuite: "processAllAvailable" |

### Gluten version

None

### PR DESCRIPTION
**Pull request by ReemaAlzaid**
This pull request upgrades the Python version from 3.9 to 3.10 in the CI workflow and enables previously disabled Arrow UDF tests for Spark 4.1.

## What changes were proposed in this pull request?

1. **CI Workflow Update**: Updated `spark-test-spark41` job in `.github/workflows/velox_backend_x86.yml` to use Python 3.10 instead of Python 3.9
2. **Test Suite Update**: Enabled ArrowEvalPythonExecSuite tests for Spark 4.1 by updating `testWithMaxSparkVersion` from "4.0" to "4.1"

## Why are the changes needed?

Fixes #11400 - Spark 4.1 requires Python 3.10 as the minimum version. The previous Python 3.9 setup was causing ArrowEvalPythonExecSuite tests to fail on Spark 4.1.

## How was this patch tested?

The changes will be tested through the CI pipeline with the updated Python 3.10 environment for Spark 4.1 tests.

### CODE REVIEW

File: backends-velox/src/test/scala/org/apache/gluten/execution/python/ArrowEvalPythonExecSuite.scala [hunk 1]
```diff
@@ -65,8 +64,7 @@ class ArrowEvalPythonExecSuite extends WholeStageTransformerSuite {
     checkAnswer(df2, expected)
   }
 
-  // TODO: fix on spark-4.1
-  testWithMaxSparkVersion("arrow_udf test: with unrelated projection", "4.0") {
+  testWithMaxSparkVersion("arrow_udf test: with unrelated projection", "4.1") {
```
Comment (baibaichen) [inline backends-velox/src/test/scala/org/apache/gluten/execution/python/ArrowEvalPythonExecSuite.scala]: use `test` directly

File: backends-velox/src/test/scala/org/apache/gluten/execution/python/ArrowEvalPythonExecSuite.scala [hunk 2]
```diff
@@ -87,8 +85,7 @@ class ArrowEvalPythonExecSuite extends WholeStageTransformerSuite {
     checkAnswer(df, expected)
   }
 
-  // TODO: fix on spark-4.1
-  testWithMaxSparkVersion("arrow_udf test: with preprojection", "4.0") {
+  testWithMaxSparkVersion("arrow_udf test: with preprojection", "4.1") {
```
Comment (baibaichen) [inline backends-velox/src/test/scala/org/apache/gluten/execution/python/ArrowEvalPythonExecSuite.scala]: ditto

**Comment by baibaichen** [comment]
We need fix
```
CentOS Linux 8 - AppStream                       32 MB/s | 8.4 MB     00:00    
CentOS Linux 8 - BaseOS                          34 MB/s | 4.6 MB     00:00    
CentOS Linux 8 - Extras                         117 kB/s |  10 kB     00:00    
CentOS Linux 8 - PowerTools                      15 MB/s | 2.3 MB     00:00    
Extra Packages for Enterprise Linux 8 - x86_64   33 MB/s |  14 MB     00:00    
Error: Problems in request:
missing groups or modules: python310
Error: Process completed with exit code 1.
```

**Comment by ReemaAlzaid** [comment]
@baibaichen replaced dnf module install python310 with dnf install python3.10
as the python310 module is not available in centOS 8

**Comment by zhouyuan** [comment]
@ReemaAlzaid the package is not avaiable on cent8
```
Extra Packages for Enterprise Linux 8 - x86_64   49 MB/s |  14 MB     00:00    
No match for argument: python3.10
No match for argument: python3.10-pip
No match for argument: python3.10-devel
Error: Unable to find a match: python3.10 python3.10-pip python3.10-devel
Error: Process completed with exit code 1.
```
